src/tp_list_maker_server.cpp

```diff
@@ -0,0 +1,377 @@
+    ROS_INFO("Ready to generate some A1 TPL's");
+    number_of_times_map_refreshed = 0;
+    //sub = n.subscribe("amcl_pose", 10, &Mratplg::current_pos_retriever, this);
+  }
+      //prototypes bb
+  bool Mratplg::generate_new_tpl(hide_and_seek::tp_map_srv::Request &req, hide_and_seek::tp_map_srv::Response &res);
+  // void current_pos_retriever(const geometry_msgs::PoseWithCovarianceStamped::ConstPtr& current_pos);
+private:
+  ros::NodeHandle n;
+  ros::ServiceServer tpl_service;
+  ros::ServiceClient map_reciever;
+  //geometry_msgs::PoseWithCovarianceStampedMessage current_position;
+  //ros::ServiceClient current_position;
+  ros::Subscriber sub;
+  int num_of_times_map_refreshed;
+};
+
+/*  
+void Mratplg::current_pos_retriever(const geometry_msgs::PoseWithCovarianceStamped::ConstPtr& current_pos){
+  this.current_position = current_pos;
+}//JUST FINISHED THIS 12:36 AM December 2nd
+*/  
+bool Mratplg::generate_new_tpl(hide_and_seek::tp_map_srv::Request &req, hide_and_seek::tp_map_srv::Response &res)
+{
+  /* if(req.postcard == true)
+  {
+    res.miniverse[req.destination].visited = true;  //"Wish you were here" :)
+    return false;
+  }*/
+  nav_msgs::GetMap get_map_srv;//no values needed since this service only gets the map B)
+  
+  //if(num_of_times_map
+  
+  if(map_reciever.call(get_map_srv))
+  {
+    OccupancyGrid map = get_map_srv.response.map;//not positive if this is completely correct
+    ++num_of_times_map_refreshed;
+  }
+  else
+  {
+    return false;
+  }
+
+  int x_min, x_max, y_min, y_max = -2;
+  //pick a new reSOLUTIon hereish?
+
+  boolean prev_check = false;
+  int miniverse_index = 0; 
+  //TODO(shaneallcroft): (after launchpad presentations) accomodate a room system
+  for (size_t x = 0; x < map.info.width; ++x) {
+    for (size_t y = 0; y < map.info.height; ++y) {
+      /* I _think_ this is right -- I'll check the GMapping source. */
+      size_t idx = map.info.width * x + y;
+      if(map.data[idx] == 0 && x_min == -2){  /* weird if it finds an isolated open space it cant get to but i dont think thats very plausible*/
+	x_min = x;
+	y_min = y;
+      }
+      else if(map.data[idx] == 0){
+	x_max = x;  //no other requirements, this is because the x and y will always be at their highest
+	y_max = y;
+      }
+    }
+  }
+  
+  res.miniverse[] = new hide_and_seek::tp_msg[ (x_max - x_min) * (y_max - y_min) ];  //tp miniverse
+  // ^^ is this legal? i dont see why it would be
+  for(int x = x_min; x <= x_max; ++x)
+  {
+    for(int y = y_min; y <= y_max; ++y)
+    {
+      /*
+	the following code iterates thru every spot on the grid in the room to see if it is occupied
+	if it is it the 8 surrounding spots on the grid are analyzed, if they are free, a tourist point is marked at the central spot first checked at the beginning of the loop
+	it could be done with a single if statement with a ton of && operators but thats so fucking ugly and this way shorts circuits anyway soooo.......
+*/
```

Comment:
``` 
       /*
          ya, this is good.
*/
```

Reply:
That comment uses python 2.0 EXCLUSIVELY so I've heard